File: samples/ios/KsDialogsSample/DeclarativeDialogViewModel.swift
import KsDialogs

/// Declarative Dialog の ViewModel。
///
/// 結果型を宣言していないため、結果は既定の真偽値になる (core/ADR-0012)。
/// この型そのものがレジストリの登録キーになる。
final class DeclarativeDialogViewModel: DialogViewModel {
    /// ダイアログに表示するメッセージ。
    let message: String

    init(message: String) {
        self.message = message
    }
}

File: samples/android/app/src/main/kotlin/jp/kamusoft/ksdialogs/samples/android/DeclarativeDialogViewModel.kt
package jp.kamusoft.ksdialogs.samples.android

import jp.kamusoft.ksdialogs.SimpleDialogViewModel

/**
 * Declarative Dialog の ViewModel。
 *
 * 結果型を書かずに済む顔で宣言しているため、結果は真偽値になる (core/ADR-0012)。
 * このクラス参照そのものがレジストリの登録キーになる。
 *
 * @property message ダイアログに表示するメッセージ
 */
internal class DeclarativeDialogViewModel(val message: String) : SimpleDialogViewModel
